src/arena.c

```diff
@@ -292,34 +292,50 @@ arena_slab_reg_alloc_batch(edata_t *slab, const bin_info_t *bin_info,
 
 static void
 arena_large_malloc_stats_update(tsdn_t *tsdn, arena_t *arena, size_t usize) {
-	szind_t index, hindex;
-
 	cassert(config_stats);
 
+	szind_t index = sz_size2index(usize);
 	if (usize < SC_LARGE_MINCLASS) {
-		usize = SC_LARGE_MINCLASS;
+		assert(index < SC_NBINS);
+		bin_t *bin = &arena->bins[index];
+		malloc_mutex_lock(tsdn, &bin->lock);
+		bin->stats.nmalloc++;
+		malloc_mutex_unlock(tsdn, &bin->lock);
+	} else {
+		szind_t hindex = (index >= SC_NBINS) ? index - SC_NBINS : 0;
```

Comment:
nit: this can be just `assert(index >= SC_NBINS); hindex = index - SC_NBINS;`. Same for the large_dalloc_stats path.

Reply:
> Same for the large_dalloc_stats path

No, this isn't correct, it doesn't handle the edge-case correctly where the bumped usize pushes you into `SC_LARGE_MINCLASS` but `usize` is less than `SC_LARGE_MINCLASS`.

Reply:
Fixed.